yt_dlp/extractor/bilibili.py

```diff
@@ -689,10 +680,29 @@ def _real_extract(self, url):
         old_video_id = format_field(aid, None, f'%s_part{part_id or 1}')
         cid = traverse_obj(video_data, ('pages', part_id - 1, 'cid')) if part_id else video_data.get('cid')
 
+        play_info = (
+            traverse_obj(
+                self._search_json(
+                    r'window\.__playinfo__\s*=', webpage, 'play info', video_id, default=None),
+                ('data', {dict}))
+            or self._download_playinfo(video_id, cid, headers=headers, fatal=False))
+        if not play_info:
+            raise ExtractorError('Failed to extract play info')
```

Comment:
this could be simplified if we leave `_download_playinfo` as always being fatal, no?